src/init.c

```diff
@@ -572,8 +614,10 @@ static void jl_resolve_sysimg_location(JL_IMAGE_SEARCH rel)
         jl_options.outputbc = abspath(jl_options.outputbc, 0);
     if (jl_options.machine_file)
         jl_options.machine_file = abspath(jl_options.machine_file, 0);
-    if (jl_options.project && strncmp(jl_options.project, "@.", strlen(jl_options.project)) != 0)
+    if (jl_options.project && (strcmp(jl_options.project, "@.") != 0 || strcmp(jl_options.project, "@") != 0))
         jl_options.project = abspath(jl_options.project, 0);
+    if (jl_options.output_code_coverage)
+        jl_options.output_code_coverage = absformat(jl_options.output_code_coverage);
```

Comment:
If you pass `jl_options.output_code_coverage` through `absformat()`, doesn't that mean that if I pass in `--code-coverage=foo.%p.info`, then this will get stored as `foo.%%p.info`, thereby negating the usefulness of the `%p`?

Reply:
no, the `absformat` call doesn't mangle the input string

Reply:
It doesn't matter; you're overwriting the pointer; `x = absformat(x)`, no?

Reply:
Yes, but that's OK because it doesn't mangle (escape) it

Reply:
I see; I had misread `absformat`; it's only escaping the `pwd()` part, not the `in` part.